proposals/simd/portable-simd.md

```diff
@@ -0,0 +1,1324 @@
+```
+
+### Square root
+* `f32x4.sqrt(a: v128, rmode: RoundingMode) -> v128`
+* `f64x2.sqrt(a: v128, rmode: RoundingMode) -> v128`
+
+Lane-wise IEEE `squareRoot`.
+
+```python
+def S.sqrt(a, rmode):
+    return S.lanewise_unary(wrap_fp_unary(ieee.squareRoot, rmode), a)
+```
+
+### Reciprocal approximation
+* `f32x4.reciprocalApproximation(a: v128) -> v128`
+* `f64x2.reciprocalApproximation(a: v128) -> v128`
+
+Implementation-dependent approximation to the reciprocal.
+
+```python
+def S.reciprocalApproximation(a):
+    def recip_approx(x):
+        if isnan(x):
+            return canonicalize_nan(x)
+        if x == 0.0:
+            # +0.0 -> +Inf, -0.0 -> -Inf.
+            return 1/x
+        if isinf(x):
+            # +Inf -> +0.0, -Inf -> -0.0.
+            return 1/x
+        # The exact nature of the approximation is unspecified.
+        return implementation_dependent(x)
+    return S.lanewise_unary(recip_approx, a)
+```
+
+### Reciprocal square root approximation
+* `f32x4.reciprocalSqrtApproximation(a: v128) -> v128`
+* `f64x2.reciprocalSqrtApproximation(a: v128) -> v128`
+
+Implementation-dependent approximation to the reciprocal of the square root.
+
+```python
+def S.reciprocalSqrtApproximation(a):
+    def recip_sqrt_approx(x):
+        if isnan(x):
+            return canonicalize_nan(x)
+        if x == 0:
+            # +0.0 -> +Inf, -0.0 -> -Inf.
+            return 1/x
+        if isinf(x):
+            # +Inf -> +0.0, -Inf -> -0.0.
+            return 1/x
+        # The exact nature of the approximation is unspecified.
+        return implementation_dependent(x)
+    return S.lanewise_unary(recip_sqrt_approx, a)
+```
+
+## Conversions
+### Integer to floating point
+* `f32x4.fromSignedInt(a: v128, rmode: RoundingMode) -> v128`
+* `f64x2.fromSignedInt(a: v128, rmode: RoundingMode) -> v128`
+* `f32x4.fromUnsignedInt(a: v128, rmode: RoundingMode) -> v128`
+* `f64x2.fromUnsignedInt(a: v128, rmode: RoundingMode) -> v128`
+
+Lane-wise conversion from integer to floating point. Some integer values will be
+rounded.
+
+```python
+def S.fromSignedInt(a, rmode):
+    def convert(x):
+        return S.LaneType.convertFromInt(x, rmode)
+    return S.lanewise_unary(convert, a)
+
+def S.fromUnsignedInt(a, rmode):
+    def convert(x):
+        return S.LaneType.convertFromInt(x, rmode)
+    return S.lanewise_unary(convert, a)
+```
+
+### Floating point to integer
+* `s32x4.fromFloat(a: v128) -> (result: v128, fail: boolean)`
+* `s64x2.fromFloat(a: v128) -> (result: v128, fail: boolean)`
+* `u32x4.fromFloat(a: v128) -> (result: v128, fail: boolean)`
+* `u64x2.fromFloat(a: v128) -> (result: v128, fail: boolean)`
+
+Lane-wise conversion from floating point to integer using the IEEE
+`convertToIntegerTowardZero` function. If any lane is a NaN or the rounded
+integer value is outside the range of the destination type, return `fail = true`
+and an unspecified `result`.
```

Comment:
Agreed. I did just that in the *Floating point conversions* section of the overview document.